src/com/intellij/idea/plugin/hybris/system/java/codeInsight/hints/LoggerInlayHintsProvider.kt

```diff
@@ -0,0 +1,178 @@
+                val range = InlayHintsUtils.getTextRangeWithoutLeadingCommentsAndWhitespaces(targetElement)
+                entries.add(range to ClickableTextCodeVisionEntry("log level", id, handler, HybrisIcons.Log.TOGGLE, "", "Setup the logger for SAP Commerce Cloud"))
+            }
+        })
+
+        return entries
+    }
+
+    fun extractIdentifierForLogger(element: PsiElement, file: PsiFile): String? = when (element) {
+        is PsiClass -> file.packageName()?.let { "$it.${element.name}" }
+        is PsiPackageStatement -> element.packageName
+        else -> null
+    }
+
+    private inner class ClickHandler(
+        element: PsiElement,
+        private val loggerIdentifier: String,
+    ) : (MouseEvent?, Editor) -> Unit {
+        private val elementPointer = SmartPointerManager.createPointer(element)
+
+        override fun invoke(event: MouseEvent?, editor: Editor) {
+            if (isInlaySettingsEditor(editor)) return
+            val element = elementPointer.element ?: return
+            handleClick(editor, element, loggerIdentifier)
+        }
+    }
+
+    fun handleClick(editor: Editor, element: PsiElement, loggerIdentifier: String) {
+        val actionGroup = DefaultActionGroup().apply {
+            add(LoggerAction("TRACE", loggerIdentifier, HybrisIcons.Log.Level.TRACE))
+            add(LoggerAction("DEBUG", loggerIdentifier, HybrisIcons.Log.Level.DEBUG))
+            add(LoggerAction("INFO", loggerIdentifier, HybrisIcons.Log.Level.INFO))
+            add(LoggerAction("WARN", loggerIdentifier, HybrisIcons.Log.Level.WARN))
+            add(LoggerAction("ERROR", loggerIdentifier, HybrisIcons.Log.Level.ERROR))
+            add(LoggerAction("FATAL", loggerIdentifier, HybrisIcons.Log.Level.FATAL))
+            add(LoggerAction("SEVERE", loggerIdentifier, HybrisIcons.Log.Level.SEVERE))
+        }
+
+        val dataManager = DataManager.getInstance()
+
+        // Show the popup menu
+        val popup = JBPopupFactory.getInstance()
+            .createActionGroupPopup(
+                "Select an Option",
+                actionGroup,
+                dataManager.getDataContext(editor.component),
+                JBPopupFactory.ActionSelectionAid.SPEEDSEARCH,
+                true
+            )
+
+        // Calculate the position for the popup
+        val offset = element.endOffset
+        val logicalPosition = editor.offsetToLogicalPosition(offset)
+        val visualPosition = editor.logicalToVisualPosition(logicalPosition)
+        val point = editor.visualPositionToXY(visualPosition)
+
+        // Convert the point to a RelativePoint
+        val relativePoint = RelativePoint(editor.contentComponent, Point(point))
+
+        // Show the popup at the calculated relative point
+        popup.show(relativePoint)
+    }
+}
+
+class LoggerAction(private val logLevel: String, val logIdentifier: String, val icon: Icon) : AnAction(logLevel, "", icon) {
```

Comment:
move this action to `BGT` thread, see https://plugins.jetbrains.com/docs/intellij/basic-action-system.html#anactiongetactionupdatethread and https://plugins.jetbrains.com/docs/intellij/threading-model.html.